src/System.Binary.Base64/System/Binary/Base64.cs

```diff
@@ -270,10 +353,12 @@ private static int DecodePadByTwo(ref byte srcBytes)
         /// <returns>Number of bytes written to the destination.</returns>
         public static int Decode(ReadOnlySpan<byte> source, Span<byte> destination)
         {
-            /*if (destination.Length < ComputeDecodedLength(source.Length))
+            int requiredLength = ComputeDecodedLength(source);
```

Comment:
Why do you check the output length here? I think we need to make the APIs to be like the text endoding APIs: they encode/decode as much as they can and then return how much they processed. This way you can implement multi-segment operations using these APIs. Otherwise, you are forcing callers to compute decoded length, then slice, then this routine computes decoded length again (for not reason), ...